client/js/components/statement/splitStatement/SegmentationEditor.vue

```diff
@@ -52,37 +52,53 @@ export default {
 
   data () {
     return {
+      customMarks: {
+        underline: {
+          parseDOM: [{ tag: 'u' }],
+          toDOM() {
+            return ['u']
+          }
+        },
+        link: {
+          attrs: {
+            href: {},
+            class: { default: null }
+          },
+          inclusive: false,
+          parseDOM: [{
+            tag: 'a[href]',
+            getAttrs (dom) {
+              return {
+                href: dom.getAttribute('href'),
+                class: dom.getAttribute('class')
+              }
+            }
+          }],
+          toDOM (node) {
+            const { href, class: className } = node.attrs
+            return ['a', { href, class: className }, 0]
+          }
+        }
+      },
       maxRange: 0
     }
   },
 
   methods: {
+   extendMarks (baseMarks, newMarks) {
+      let extendedMarks = baseMarks
+
+      for (const [key, value] of Object.entries(newMarks)) {
+        extendedMarks = extendedMarks.update(key, value)
```

Comment:
Where does .update come from?

Reply:
It comes from the external library Immutable.js, but I think it was a bad idea to use this method. However, I could not find another way to extend the marks with the custom ones, except by adding all the necessary tags directly to the marks..

Reply:
I'm confused. Aren't the `schema.spec.marks` now missing? It seems like there are not all marks included (eg. em, strong, code). Does it not work with the spread operator?

Reply:
No, the spread operator does not work, and the default marks from ProseMirror are overwritten when using the spread operator